### DISCUSSION TITLE
Instructions to use Mojo on Mac with remote Ubuntu server and Docker

### DISCUSSION THREAD

**Discussion by tom-rigelblu** [modular/modular#968] [Show and tell]
It took me a bit to figure out how to use Mojo on remote Ubuntu server with Docker, so I wrote up instructions how I made it work

# tl;dr
**Server:**
- Docker on a remote server listens to connections from other computers

**Client:**
- Watch for file changes in the personal computer's project src dir
- Sync file changes to the remote server
- Tell Docker on your personal computer to use (i.e. execute) the Docker on the remote server

**Limitations:** rsync is used because Docker cannot bind a dir/mount a volume to a remote server

It builds on top of
- [Modular Docker Example](https://github.com/modularml/mojo/tree/main/examples/docker)
- [Using Mojo🔥 with Docker containers](https://youtu.be/cHyYmF-RhUk?si=LwyNAobjKlCpdjLz)

Step by step instruction on: [mojo-remote-docker](https://github.com/tom-rigelblu/mojo-remote-docker)

**Comment by tom-rigelblu** [comment]
I realize the mojo vscode extension only works if the mojosdk is installed locally. So you need to use the docker "Attach Visual Studio Code" feature and work on the code remotely (rather than locally).

**Reply by BlackHo0d** [reply]
You work on Mac(Intel) or Mac(M1) ?

**Reply by tom-rigelblu** [reply]
M1
